arangod/IResearch/IResearchLinkMeta.cpp

```diff
@@ -475,6 +535,33 @@ bool FieldMeta::json(ArangodServer& server, velocypack::Builder& builder,
     fieldsBuilder.close();
     builder.add("fields", fieldsBuilder.slice());
   }
+#ifdef USE_ENTERPRISE
+  if (!_nested.empty()) {
+    velocypack::Builder fieldsBuilder;
+    Mask fieldMask(true);      // output all non-matching fields
+    auto subDefaults = *this;  // make modifable copy
+    fieldsBuilder.openObject();
+
+    for (auto& entry : _nested) {
+      fieldMask._fields =
+          !entry.value()
+               ->_fields.empty();  // do not output empty fields on subobjects
+      fieldsBuilder.add(           // add sub-object
+          std::string_view(entry.key().c_str(),
+                           entry.key().size()),  // field name
```

Comment:
what is key? maybe we have some nicer way?

Reply:
This is how UnorderedRefKeyMap iterator implemented. Keys here are field names, so we used them as attribute names in definition json